internal/compiler/object_tree.rs

```diff
@@ -589,6 +594,30 @@ impl GeometryProps {
 
 pub type BindingsMap = BTreeMap<String, RefCell<BindingExpression>>;
 
+#[derive(Clone)]
+pub struct ElementDebugInfo {
+    // The id qualified with the enclosing component name. Given `foo := Bar {}` this is `EnclosingComponent::foo`
+    pub qualified_id: Option<String>,
+    pub type_name: String,
```

Comment:
Exactly because of that this patch does not use the AST. It should be Button, not MyBytton. That’s why the code uses the ElementType’s type_name(): https://github.com/slint-ui/slint/blob/10698113fe7396995ad3e6f38e8f503c91bab502/internal/compiler/object_tree.rs#L942